akka-actor/src/main/scala/akka/pattern/AskSupport.scala

```diff
@@ -553,7 +554,11 @@ private[akka] final class PromiseActorRef private (
       if (updateState(null, Registering)) {
         var p: ActorPath = null
         try {
-          p = provider.tempPath()
+          val prefix = targetName match {
+            case ref: ActorRef => ref.path.name
+            case other         => other.toString.replaceAll("[./*]", "_")
```

Comment:
what other can it be? ActorSelection? then I think we need something better than `toString`

Reply:
It can be:


* `ActorSelection` for classic selection ask - we'd want to escape and use the selection expression?
* `InternalRecipientRef` in typed (In practice isn't always an `ActorRef` **except** when it is from `EntityRefImpl`) - not sure what we can do for these since the types are in akka-actor-typed
* the string "unknown", through `AskPromiseRef.apply`, which is public API (!?) - just use as is

Reply:
Perhaps better to introduce a separate field for the prefix instead of tightly coupling the PromiseRef with all ways it can be used.